File: A2SV/Feburary/leetcode/count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int n = nums.size();
        int ans = 0;
        int pre = 0;
        vector<int> cnt(n + 1, 0);
        cnt[0] = 1;
        for (int i = 0; i < n; i++)
        {
            pre += (nums[i] & 1);
            if (pre >= k)
            {
                ans += cnt[pre - k];
            }
            cnt[pre] += 1;
        }
        return ans;
    }
};

File: A2SV/Feburary/leetcode/count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        ans = 0
        pre = 0
        cnt = [0] * (n + 1)
        cnt[0] = 1
        for i in range(n):
            pre += nums[i] & 1
            if pre >= k:
                ans += cnt[pre - k]
            cnt[pre] += 1
        return ans